presto-hive/src/main/java/com/facebook/presto/hive/PartitionLoader.java

```diff
@@ -35,14 +38,19 @@
     public abstract ListenableFuture<?> loadPartition(HivePartitionMetadata partition, HiveSplitSource hiveSplitSource, boolean stopped)
             throws IOException;
 
-    public List<HivePartitionKey> getPartitionKeys(Table table, Optional<Partition> partition)
+    public List<HivePartitionKey> getPartitionKeys(Table table, Optional<Partition> partition, String partitionName)
     {
         if (!partition.isPresent()) {
             return ImmutableList.of();
         }
         ImmutableList.Builder<HivePartitionKey> partitionKeys = ImmutableList.builder();
+        // partition information provided by Hive Metastore is out of order
```

Comment:
partition information provided by Hive Metastore could be out of order

Reply:
Ok, let me modify it